modules/services/unison.nix

```diff
@@ -0,0 +1,127 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta.maintainers = with maintainers; [ pacien ];
+
+  options.services.unison = let
+    pairOf = t: with types; addCheck (listOf t) (l: length l == 2);
+    pairOptions = {
+      options = {
+        stateDirectory = mkOption {
+          type = types.path;
+          default = "${config.xdg.dataHome}/unison";
+          defaultText = "$XDG_DATA_HOME/unison";
+          description = ''
+            Unison state directory to use.
+          '';
+        };
+
+        commandOptions = mkOption rec {
+          type = with types; attrsOf str;
+          apply = mergeAttrs default;
+          default = {
+            sshcmd = "${pkgs.openssh}/bin/ssh";
+            ui = "text";
+            auto = "true";
+            batch = "true";
+            log = "false"; # don't log to file, handled by systemd
```

Comment:
Better always set these options?

Reply:
Those are applied by default by `apply = mergeAttrs default`.

I'm not sure whether these values should be set in `config` instead, but `types.attrsOf` doesn't seem to play well with `mkDefault`.

Reply:
I would suggest setting them directly, not via the option. I.e., to have something like
```
serialiseArgs (defaultCommandOpts // pairCfg.commandOptions)
```
where `defaultCommandOpts` contains these values. Then they always will be set unless the user explicitly overrides them.

Reply:
The effect would be the same as with `apply` but the defaults would not appear in the documentation anymore. Is that desirable?

Reply:
Hmm, I'm not entirely certain, my intuition is that the `apply` trick could cause issues when merging options. But I can't think of a particular case where it could cause issues. I'll have to think about it 🙂

Reply:
Note: https://github.com/NixOS/rfcs/pull/42 proposes new types that merge properly ([unlike `types.attrs`](https://github.com/NixOS/nixpkgs/pull/63589#discussion_r300768976)) and the inclusion of default values specified in `config` in the generated documentation.